dbms/src/Interpreters/InterpreterKillQueryQuery.cpp

```diff
@@ -179,59 +182,106 @@ BlockIO InterpreterKillQueryQuery::execute()
         return executeDDLQueryOnCluster(query_ptr, context, {"system"});
 
     BlockIO res_io;
-    Block processes_block = getSelectFromSystemProcessesResult();
-    if (!processes_block)
-        return res_io;
+    switch (query.type)
+    {
+    case ASTKillQueryQuery::Type::Query:
+    {
+        Block processes_block = getSelectResult("query_id, user, query", "system.processes");
+        if (!processes_block)
+            return res_io;
+
+        ProcessList & process_list = context.getProcessList();
+        QueryDescriptors queries_to_stop = extractQueriesExceptMeAndCheckAccess(processes_block, context);
+
+        auto header = processes_block.cloneEmpty();
+        header.insert(0, {ColumnString::create(), std::make_shared<DataTypeString>(), "kill_status"});
+
+        if (!query.sync || query.test)
+        {
+            MutableColumns res_columns = header.cloneEmptyColumns();
 
-    ProcessList & process_list = context.getProcessList();
-    QueryDescriptors queries_to_stop = extractQueriesExceptMeAndCheckAccess(processes_block, context);
+            for (const auto & query_desc : queries_to_stop)
+            {
+                auto code = (query.test) ? CancellationCode::Unknown : process_list.sendCancelToQuery(query_desc.query_id, query_desc.user, true);
 
-    auto header = processes_block.cloneEmpty();
-    header.insert(0, {ColumnString::create(), std::make_shared<DataTypeString>(), "kill_status"});
+                /// Raise exception if this query is immortal, user have to know
```

Comment:
Obsolete comment.

Reply:
It is not only obsolete but plain wrong.